video/out/vulkan/utils.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+#include "video/out/vo.h"
+#include "video/out/gpu/context.h"
+#include "video/mp_image.h"
+
+#include "common.h"
+#include "formats.h"
+
+#define VK_LOAD_PFN(name) PFN_##name pfn_##name = (PFN_##name) \
+                            vkGetInstanceProcAddr(vk->inst, #name);
+
+// Return a human-readable name for various struct mpvk_ctx enums
+const char* vk_err(VkResult res);
+
+// Convenience macros to simplify a lot of common boilerplate
+#define VK_ASSERT(res, str)                               \
+    if (res != VK_SUCCESS) {                              \
+        MP_ERR(vk, str ": %s\n", vk_err(res));            \
+        goto error;                                       \
+    }
```

Comment:
Use the `do { ... } while(0)` idiom for all these macros (including the one below) to avoid surprises.